src/auro-select.js

```diff
@@ -216,27 +216,41 @@ export class AuroSelect extends LitElement {
   }
 
   /**
-   * Adds the value string to the trigger.
-   * @param {string} str - The string to display in the trigger.
+   * Adds the value to the trigger.
+   * @param {object} option - The selected option to display it's value.
    * @private
    * @returns {void}
    */
-  updateDisplayedValue(str) {
+  updateDisplayedValue(option) {
     const dropdown = this.shadowRoot.querySelector('auro-dropdown');
     const triggerContentEl = dropdown.querySelector('#triggerFocus');
 
     // remove all existing rendered value(s)
+    triggerContentEl.querySelectorAll('auro-menuoption').forEach((elm) => {
+      elm.remove();
+    });
+
     triggerContentEl.querySelectorAll('[valuestr]').forEach((elm) => {
       elm.remove();
     });
 
-    // create a new element with the new value
-    const valueElem = document.createElement('span');
-    valueElem.setAttribute('valuestr', true);
-    valueElem.appendChild(document.createTextNode(str));
+    if (typeof option === 'string') {
+      // create a new span element with the value string
+      const valueElem = document.createElement('span');
+      valueElem.setAttribute('valuestr', true);
+      valueElem.appendChild(document.createTextNode(option));
```

Comment:
```suggestion
     valueElem.textContent = option;
```

You could use the `textContent` property to optimize this function.